setup.py

```diff
@@ -89,6 +90,10 @@ def compute_num_jobs(self):
                 num_jobs = len(os.sched_getaffinity(0))
             except AttributeError:
                 num_jobs = os.cpu_count()
+            physical_ram_gb = psutil.virtual_memory().total / (1024**3)
+            # Set the number of jobs such that each job gets about 8GB of RAM
+            # to better avoid OOM during compilation.
+            num_jobs = min(num_jobs, int(max(physical_ram_gb // 8, 1)))
```

Comment:
In my experience, I've seen flash attention 3 requiring up to 17/18 GB of memory per job, so I'm not sure using 8 will be enough.